paywall/src/__tests__/hooks/browser/useListenForPostMessage.test.js

```diff
@@ -181,6 +191,78 @@ describe('useListenForPostMessage hook', () => {
 
         expect(wrapper.getByTitle('value')).toHaveTextContent('got it!')
       })
+
+      it('calls the getValue helper to merge the value with its defaults', () => {
+        expect.assertions(1)
+
+        let wrapper
+        rtl.act(() => {
+          wrapper = rtl.render(
+            <Wrapper
+              defaultValue="boo"
+              getValue={(v, d) => JSON.stringify(v) + d}
+            />
+          )
+        })
+
+        // this is the callback saveData, which was passed to addEventListener
+        const listener = getListener()
+
+        rtl.act(() => {
+          listener(event)
+        })
+
+        expect(wrapper.getByTitle('value')).toHaveTextContent(
+          '"{\\"thing\\":\\"got it!\\"}boo"'
+        )
+      })
+
+      it('updates the value only once if it does not change', () => {
+        expect.assertions(4)
+        called = jest.fn()
+        let listener
+
+        const listeners = {
+          message: new Map(),
+        }
+
+        fakeWindow.addEventListener = (event, cb) => {
+          listener = cb
+          listeners.message.set(cb, cb)
+        }
+        fakeWindow.removeEventListener = (event, cb) => {
+          listeners.message.delete(cb)
+        }
+
+        let wrapper
+        rtl.act(() => {
+          wrapper = rtl.render(<Wrapper defaultValue="hi" />)
+        })
+        expect(wrapper.getByTitle('value')).toHaveTextContent('hi')
+
+        rtl.act(() => {
+          listener(event)
+        })
+        expect(wrapper.getByTitle('value')).toHaveTextContent('got it!')
+        event.data.payload = { ...event.data.payload }
+
+        rtl.act(() => {
+          listener(event)
+        })
+        event.data.payload = { ...event.data.payload }
```

Comment:
I think it would be worthwhile to adds some comments here explaining why `rtl.act` is being called a number of times within the test

Reply:
ok